### PR TITLE
New tutorial: Detecting payment card fraud with Temporian and TensorFlow Decision Forests

### PR DESCRIPTION
**Pull request by achoum**

### CODE REVIEW

File: docs/src/tutorials/bank_fraud_detection_with_tfdf.ipynb [hunk 1]
```diff
@@ -0,0 +1,1316 @@
...
+    "\n",
+    "Detection of fraud in online banking is critical for banks, businesses, and their consumers. The book \"[Reproducible Machine Learning for Credit Card Fraud Detection](https://fraud-detection-handbook.github.io/fraud-detection-handbook/Foreword.html)\" by Le Borgne et al. introduces the problem of payment card fraud and shows how fraud can be detected using machine learning. However, since banking transactions are sensitive and not widely available, the book uses a synthetic dataset for practical exercises.\n",
+    "\n",
+    "This notebook uses the same dataset to show how to use Temporian and [TensorFlow Decision Forests](https://www.tensorflow.org/decision_forests) to detect fraud. Temporian is used for data preprocessing and augmentation, while TensorFlow Decision Forests is used for model training. Data augmentation is often critical for temporal data, and this notebook demonstrates how complex data augmentation can be performed with ease using Temporian.\n",
+    "\n",
+    "The notebook is divided into three parts:\n",
+    "\n",
+    "1. Download the dataset and load it as a Temporian EventSet.\n",
```
Comment (rstz) [inline docs/src/tutorials/bank_fraud_detection_with_tfdf.ipynb:19]: Download the dataset and import it to Temporian

File: docs/src/tutorials/bank_fraud_detection_with_tfdf.ipynb [hunk 2]
```diff
@@ -0,0 +1,1316 @@
...
+   "source": [
+    "dataset_tp.plot()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2477a90d-6ddd-4da8-91e8-9b7cae085537",
+   "metadata": {},
+   "source": [
+    "Instead, we can plot the transaction of a single client."
```
Comment (rstz) [inline docs/src/tutorials/bank_fraud_detection_with_tfdf.ipynb:487]: We can also plot the transaction of  a single client

File: docs/src/tutorials/bank_fraud_detection_with_tfdf.ipynb [hunk 3]
```diff
@@ -0,0 +1,1316 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b2f4a4fb-fc01-48a8-b69c-6fdf96ea65eb",
+   "metadata": {},
+   "source": [
+    "After exploring the dataset, we want to compute some augmented features that may correlate with fraudulent activities. We will compute the following three features:\n",
+    "\n",
+    "**Calendar features**: We will extract the hour of the day and the day of the week as individual features. This is because fraudulent transactions may be more likely to occur at specific times.\n",
```
Comment (rstz) [inline docs/src/tutorials/bank_fraud_detection_with_tfdf.ipynb:521]: **Calendar features**: Extract the hour of the day and the day of the week as individual features. This is useful because fraudulent transactions may be more likely to occur at specific times.

File: docs/src/tutorials/bank_fraud_detection_with_tfdf.ipynb [hunk 4]
```diff
@@ -0,0 +1,1316 @@
...
+    "\n",
+    "**Moving sum of fraud per customer**: For each client, we will extract the number of fraudulent transactions in the last 4 weeks. This is because clients who start to commit fraud (maybe the a card was stolen) may be more likely to commit fraud in the future. However, since we only know after a week if a transaction is fraudulent, there will be a lag in this feature.\n",
```
Comment (rstz) [inline docs/src/tutorials/bank_fraud_detection_with_tfdf.ipynb:523]: This is because --> This is useful because

File: docs/src/tutorials/bank_fraud_detection_with_tfdf.ipynb [hunk 5]
```diff
@@ -0,0 +1,1316 @@
...
...nLEb5WtZr4tpfpdEW2RsB17YvB/nkLn79llRLdzYCUzrH2Tplb99f1rqLSExMaP9Pst+A7Do3laPqjhoPeu8EzAIVm8Ko8jWeTol0R2tP12cbHNU+XozRdlgX48ngis75ckt7AW/5ZHf1q6vq2CTr54VA6oXq06hVXlTev9TUVGDTx47jtMmvAWekyKJjOt1YIx5+5TNAaFjHb96u57VsVwg2GhYDbF8GwAQEWvhO6xsjgBF3AJcukL/WWf2/bSlfH333pNd8T2pqqnf3TRLy6G5ZVut39GovaRHPmLkIHtkEhITw311pk3fGfTigUae+pqYGd911F3bv3o1ffvkFa9aswZAhQ/Dmm2/ioosu0iIKpKamYsMG+7Uj6enp2LRpkyZxsKC4uFjdbqWqDNOTCqf9GutUJwfrzIVKvdcw4Gj76wN7DesY/XfkJF1u0Ghciba/SmOveKdSta/ScHddrZqGkLcdj1GectZVoeWzB2ABnHfgOnlnT3vkykrH/6p9jRRv3gcjd141wiXdhUZhY61xBnG1RHWDVBu/77bNAzo1BlXmU7rLfXUpX04u/yc/Q8dZGrXKi8r7V1xcjCy1cfqKn9cznW76xeIDB+Xt3dtPkbWIT7CXpe2dvGkfA90z+NdqvjcJKN/Fn+dqx1noPA6e4VWfWlxcjKxuLd7vEPtZJ9UhZrO8j1ex101nnSGkSa6ys7ORlJSEbdu2YcSIEXjhhRfw7rvv4s4778Qdd9yhRRSdFmHnfqc4M0xPphWJr133gOMwxFNZzhsH9L2Y/5x3ofbTWrwxVcpsBsxme93bj3sFNdN//MHxtE+vs9SVykwBbW/UblvCV8KaTn31ladAHiAuK5/fz38Am3Kj2teIcfk+KGitdsPDTjg1ziXdhQZo9hRj7x/g7tRJNVNANfT7btm8gLd9siv5FPbMiGzfLygymf9etU9dGrXKi9L9GzzDxh4CAwPVxanHelctpvnqvS7XzTXiHtk7awS7mP4Z8MAffIceALpn8t+nf8p/d3dpTl6+V32qbtp7va2oV3tJm3h82ua9gGZr6p999lkEBQVZj11wwQXYuXMnAgICtIii06K0qZ9LeFLhCNceL+TXlzed5f8e3yMfhrhSHzOXf8p90yreMWrpJL1ciWqiu1p8dIMvzWlvqDTd9j/bhsrR7R2N2x3L+fO06uwZdV2yloj
```
Comment (rstz) [inline docs/src/tutorials/bank_fraud_detection_with_tfdf.ipynb:664]: Save the Temporian program that computes the augmented transactions to disk. 

The second sentence is a bit unclear - what does "included with the model" mean? Is "included in the ML pipeline" better?

Comment (rstz) [inline docs/src/tutorials/bank_fraud_detection_with_tfdf.ipynb:666]: A saved Temporian program can also be applied on a large dataset that does not fit in memory using the [Apache Beam](https://beam.apache.org/) backend.

Comment (rstz) [inline docs/src/tutorials/bank_fraud_detection_with_tfdf.ipynb:716]: I'd shorten this to "Plot the relation between the augmented features and the label" - otherwise it seems like you're always converting back-and-forth between temporian and pandas.

Comment (rstz) [inline docs/src/tutorials/bank_fraud_detection_with_tfdf.ipynb:1291]: The current AUC is never explicitly said.

File: docs/src/tutorials/bank_fraud_detection_with_tfdf.ipynb [hunk 6]
```diff
@@ -0,0 +1,1316 @@
...
+    }
+   ],
+   "source": [
+    "import os\n",
+    "import datetime\n",
+    "import concurrent.futures\n",
+    "\n",
+    "import pandas as pd\n",
+    "import temporian as tp\n",
+    "import matplotlib.pyplot as plt\n",
```
Comment (rstz) [inline docs/src/tutorials/bank_fraud_detection_with_tfdf.ipynb:94]: As above, split the imports into those you actually need and those you only need for plotting / evaluation (plt, sns and sk_metrics). Otherwise it looks like you're creating a lot of dependencies.

File: docs/src/tutorials/bank_fraud_detection_with_tfdf.ipynb [hunk 7]
```diff
@@ -0,0 +1,1316 @@
...
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "def load_date(filename):\n",
+    "    print(\".\",end=\"\", flush=True)\n",
+    "    return pd.read_pickle(f\"https://github.com/Fraud-Detection-Handbook/simulated-data-raw/raw/main/data/{filename}.pkl\")\n",
+    "\n",
+    "print(\"Downloading dataset\",end=\"\")\n",
+    "with concurrent.futures.ThreadPoolExecutor(max_workers=2) as executor:\n",
```
Comment (rstz) [inline docs/src/tutorials/bank_fraud_detection_with_tfdf.ipynb:370]: Is this really necessary?

**Review by rstz — APPROVED** [review]
It's great, very clear and understandable :)

File: docs/src/tutorials/bank_fraud_detection_with_tfdf.ipynb (hunk 1 above)
Comment (achoum) [reply docs/src/tutorials/bank_fraud_detection_with_tfdf.ipynb:19]: *Done

File: docs/src/tutorials/bank_fraud_detection_with_tfdf.ipynb (hunk 3 above)
Comment (achoum) [reply docs/src/tutorials/bank_fraud_detection_with_tfdf.ipynb:521]: *Done

File: docs/src/tutorials/bank_fraud_detection_with_tfdf.ipynb (hunk 5 above)
Comment (achoum) [reply docs/src/tutorials/bank_fraud_detection_with_tfdf.ipynb:666]: *Improve

File: docs/src/tutorials/bank_fraud_detection_with_tfdf.ipynb (hunk 6 above)
Comment (achoum) [reply docs/src/tutorials/bank_fraud_detection_with_tfdf.ipynb:94]: Good point.
*Done

File: docs/src/tutorials/bank_fraud_detection_with_tfdf.ipynb (hunk 7 above)
Comment (achoum) [reply docs/src/tutorials/bank_fraud_detection_with_tfdf.ipynb:370]: Otherwise, the dataset takes a long time to load. Increase the number of threads to 4 (I got throttled by github during development).

**Comment by achoum** [comment]
Thanks!
